src/plugins/intel_gna/src/transformations/insert_copy_layer.cpp

```diff
@@ -157,8 +157,7 @@ InsertCopyBeforeAssignLayer::InsertCopyBeforeAssignLayer() {
 
             // Crop -> Memory, Input -> Split -> Memory, Concat -> Memory
             if ((std::dynamic_pointer_cast<ngraph::op::CropIE>(current_node) && !is_crop_affined(current_node)) ||
-                std::dynamic_pointer_cast<ngraph::opset8::Concat>(current_node) ||
-                std::dynamic_pointer_cast<ngraph::opset8::Split>(current_node) ||
+                is_concat(current_node) || std::dynamic_pointer_cast<ngraph::opset8::Split>(current_node) ||
```

Comment:
don't you want to add is_split()?

Reply:
No, there are tons of this kind of check. I have added is_concat, because I was using it in my code, then changed all occurrences of std::dynamic_pointer_cast<ngraph::opset8::Split>() in the rest of the code. 
Changing all other checks for other nodes is outside of scope of this task.